### PR TITLE
fix: different LooksLikeFormat algorithm

### LINKED ISSUE #18: LooksLikeFormat fails incorrectly
I have a jpg image that fails the `LooksLikeFormat` test.

In particular is fails on this check `data[l-2] != 0xff`  I'd be happy to share the image to help debug.

The image was captured with a FUJIFILM FI019

### PR DESCRIPTION
**Pull request by neurosnap**
This implementation is the one used by `http.DetectContentType` found here: https://cs.opensource.google/go/go/+/master:src/net/http/sniff.go;l=126;drc=8b364451e2e2f2f816ed877a4639d9342279f299

Fixes #18

### CODE REVIEW

File: media_parser_test.go [hunk 1]
```diff
@@ -240,14 +240,19 @@ func TestJpegMediaParser_ParseBytes_MultipleEois(t *testing.T) {
...
+		data, err := ioutil.ReadFile(filepath)
+		log.PanicIf(err)
 
-	jmp := NewJpegMediaParser()
+		jmp := NewJpegMediaParser()
 
-	if jmp.LooksLikeFormat(data) != true {
-		t.Fatalf("not detected as JPEG")
+		if jmp.LooksLikeFormat(data) != true {
+			t.Fatalf("(%s) not detected as JPEG", filepath)
```
Comment (dsoprea) [inline media_parser_test.go]: Typically in-literal parentheses are used for numbers and square-brackets are used for strings. Helps us emphasize when a value is empty.

**Review by dsoprea — CHANGES_REQUESTED** [review]

File: media_parser_test.go (hunk 1 above)
Comment (neurosnap) [reply media_parser_test.go]: Updated!

**Comment by neurosnap** [comment]
Ping on this PR.

**Comment by dsoprea** [comment]
The need for this bugs me. I wish we better understood the Fuji format. Typically, the EOI segment should be the last in the file, but some cameras/facilities take advantage of this to extend the structure/data in ways that standards-oriented clients won't notice.

**Comment by dsoprea** [comment]
Since there are no import changes, I removed the modules changes from the PR.

**Pull request merged**